### PR TITLE
Fix updatecli update-beats pipeline

### PR DESCRIPTION
**Pull request by reakaleek**
## Motivation/summary

The updatecli pipeline "update-beats" is currently failing with the error message:

```
ERROR: something went wrong in target "beats" : "no changed file to commit"
```

Source: https://github.com/elastic/apm-server/actions/runs/4344796011

This happens when a PR already exists.

This change skips the target execution if a PR already exists and its value is already up-to-date.

## How to test these changes

Execute the following command multiple times:

```shell
BRANCH_NAME=7.17 GIT_USER=ok GIT_EMAIL=ok@ok.com GITHUB_TOKEN=<your_personal_access_token> updatecli -c .ci/update-beats.yml apply
```

it should not fail anymore.

I personally tested it with the following diff on my fork:

```diff
scms:
  default:
    kind: github
    spec:
      user: '{{ requiredEnv "GIT_USER" }}'
      email: '{{ requiredEnv "GIT_EMAIL" }}'
-     owner: elastic
+     owner: reakaleek 
      repository: apm-server
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "GIT_USER" }}'
      branch: '{{ requiredEnv "BRANCH_NAME" }}'
```

### CODE REVIEW

File: .ci/update-beats.yml [hunk 1]
```diff
@@ -37,6 +37,21 @@ sources:
     spec:
       file: 'https://api.github.com/repos/elastic/beats/commits?sha={{ requiredEnv "BRANCH_NAME" }}&per_page=1'
       key: '.[0].sha'
+    transformers:
+      # substring 12 chars so it works for the condition
```
Comment (v1v) [inline .ci/update-beats.yml:41]: Thanks for this!

File: .ci/update-beats.yml [hunk 2]
```diff
@@ -37,6 +37,21 @@ sources:
...
+      - findsubmatch:
+          pattern: ^(.{12}).*
+          captureindex: 1
+
+conditions:
+  is:
+    kind: file
```
Comment (v1v) [inline .ci/update-beats.yml:48]: NIT:

What do you think adding the name field to help with the messages in the PR description?

```suggestion
    name: Is version 'github.com/elastic/beats@{{ source "beats" }}' not updated in 'go.mod'?
    kind: file
```

For instance, this [run](https://github.com/elastic/golang-crossbuild/actions/runs/4354393920/jobs/7609633045#step:3:467) uses https://github.com/elastic/golang-crossbuild/blob/7971772e9dd1b54c660b1c32cabbb607c970621d/.ci/bump-golang.yml#L64

**Review by v1v — APPROVED** [review]
LGTM with a tiny comment

File: .ci/update-beats.yml (hunk 2 above)
Comment (reakaleek) [reply .ci/update-beats.yml:49]: https://github.com/elastic/apm-server/pull/10436/commits/b36deef88372ba8a77dc85ea27312fdc7c05914f

**Review by axw — APPROVED** [review]
Thank you!

**Pull request merged**